worlds/yoshisisland/Locations.py

```diff
@@ -0,0 +1,351 @@
+from typing import List, Tuple, Optional, Callable, NamedTuple
+from BaseClasses import MultiWorld
+from .level_logic import YoshiLogic
+
+
+class LocationData(NamedTuple):
+    region: str
+    name: str
+    code: Optional[int]
+    LevelID: int
+    rule: Callable = lambda state: True
+
+
+def get_locations(multiworld: Optional[MultiWorld], player: Optional[int], boss_order: Optional[list], luigi_pieces: Optional[int], world) -> Tuple[LocationData, ...]:

```

Comment:
Change to just pass in world instead of everything

Reply:
Also you're showing -> Tuple[LocationData, ...] but the return value is a List[LocationData]. Please choose one or the other.

Reply:
Now you've changed the return to a tuple, but the location_table itself is a list, so there's still a mismatch here.
I think the only thing left here is to change the [] into () for the location_table, then instead of it being tuple(location_table) at the end, it's just location_table